src/revChatGPT/ChatGPT.py

```diff
@@ -23,26 +27,63 @@ def __init__(self, config, conversation_id=None, parent_id=None) -> None:
         self.session = tls_client.Session(
             client_identifier="chrome_105"
         )
-        self.session.cookies.set(
-            "__Secure-next-auth.session-token", config["session_token"])
         if "proxy" in config:
+            if type(config["proxy"]) != str:
+                raise Exception("Proxy must be a string!")
             proxies = {
                 "http": config["proxy"],
                 "https": config["proxy"],
             }
             self.session.proxies.update(proxies)
-        self.get_cf_cookies()
+        if "verbose" in config:
+            if type(config["verbose"]) != bool:
+                raise Exception("Verbose must be a boolean!")
+            self.verbose = config["verbose"]
+        else:
+            self.verbose = False
+        self.conversation_id = conversation_id
+        self.parent_id = parent_id
+        self.conversation_id_prev_queue = []
+        self.parent_id_prev_queue = []
+        self.isMicrosoftLogin = False
+        self.twocaptcha_key = None
+        # stdout colors
+        self.GREEN = '\033[92m'
+        self.WARNING = '\033[93m'
+        self.ENDCOLOR = '\033[0m'
+        if "email" in config and "password" in config:
+            if type(config["email"]) != str:
+                raise Exception("Email must be a string!")
+            if type(config["password"]) != str:
+                raise Exception("Password must be a string!")
+            self.email = config["email"]
+            self.password = config["password"]
+            if "isMicrosoftLogin" in config and config["isMicrosoftLogin"] == True:
+                self.isMicrosoftLogin = True
+                self.microsoft_login()
+            elif "captcha" in config:
+                if type(config["captcha"]) != str:
+                    raise Exception("2Captcha API Key must be a string!")
+                self.twocaptcha_key = config["captcha"]
+                self.email_login(self.solve_captcha())
+            else:
+                raise Exception("Invalid config!")
+        elif "session_token" in config:
+            if type(config["session_token"]) != str:
+                raise Exception("Session token must be a string!")
+            self.session_token = config["session_token"]
+            self.session.cookies.set(
+                "__Secure-next-auth.session-token", config["session_token"])
+            self.get_cf_cookies()
+        else:
+            raise Exception("Invalid config!")
```

Comment:
I see session token is still available as an option. I will update the wiki to allow users to choose.

Reply:
Yes, so it's either:

- Session Token
- Email + Password + 2Captcha Key
- Email + Password for Microsoft Login
as shown in https://github.com/skippyyy/ChatGPT/blob/main/wiki/Setup.md

Reply:
It's shown in the wiki but not readme which is a bit confusing. Will need to fix that

Reply:
Or maybe just remove the config from readme and force users to go to wiki

Reply:
done